### PR TITLE
feat: add terraform + multipass conf

### LINKED ISSUE #33: Use a hypervisor supported on all major operating systems
MacOS users reported that VirtualBox is not compatible with the latest Apple M1/M2 processors.

Using [Multipass](https://multipass.run/) seems to be a viable option.

### PR DESCRIPTION
**Pull request by Nuttymoon**
### Linked issues

- Fixes #33 

### Changes

- Make most of the Ansible var files dynamic to adapt to changing IP addresses

### Breaking changes

- Vagrant+VirtualBox removed in favor of Terraform+Multipass

### Additional comments

Because allocating static IPs in Multipass is a pain in the ass (see [How to configure static IPs](https://multipass.run/docs/configure-static-ips)), I decided to template the Ansible `hosts` file with Terraform (`inventories/local/hosts` is now a symlink to `terraform/multipass/hosts.ini` that will be created upon `terraform apply`).

I am using the example SSH key provided at [Configure instance to be managed by Ansible](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#configure-instance-to-be-managed-by-ansible). I think this is acceptable for local environments. If we want a cleaner option, we could leverage the Terraform [tls_private_key provider](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key).

To try this out:
```bash
# Install Multipass. See https://multipass.run/install
# Install Terraform. See https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli

# Initialize the Terraform project
terraform -chdir=terraform/multipass init
# Create the local instances through Multipass
terraform -chdir=terraform/multipass apply
# Bootstrap the local environment
ansible-playbook ash.avalanche.bootstrap_local_network -i inventories/local
```

### CODE REVIEW

**Review by leopaul36 — CHANGES_REQUESTED** [review]
Getting:

```
fatal: [validator01]: UNREACHABLE! => changed=false
  msg: |-
    Failed to connect to the host via ssh: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
    @         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
    Permissions 0640 for './files/multipass/id_rsa' are too open.
    It is required that your private key files are NOT accessible by others.
    This private key will be ignored.
    Load key "./files/multipass/id_rsa": bad permissions
    ansible@10.252.190.144: Permission denied (publickey).
  unreachable: true
```

We should probably commit `files/multipass/id_rsa` with `600` permissions

**Comment by leopaul36** [comment]
Works fine otherwise :+1: , seems a lot faster than Vagrant / VBox

**Comment by Nuttymoon** [comment]
> Getting:
> 
> ```
> fatal: [validator01]: UNREACHABLE! => changed=false
>   msg: |-
>     Failed to connect to the host via ssh: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
>     @         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
>     @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
>     Permissions 0640 for './files/multipass/id_rsa' are too open.
>     It is required that your private key files are NOT accessible by others.
>     This private key will be ignored.
>     Load key "./files/multipass/id_rsa": bad permissions
>     ansible@10.252.190.144: Permission denied (publickey).
>   unreachable: true
> ```
> 
> We should probably commit `files/multipass/id_rsa` with `600` permissions

We cannot version with other permission than `644` or `755`, see [Git index format docs](https://git-scm.com/docs/index-format):
> 9-bit unix permission. Only 0755 and 0644 are valid for regular files.

I will add a chmod operation in the `setup.sh`.

File: inventories/local/group_vars/avalanche_nodes.yml [hunk 1]
```diff
@@ -19,3 +20,4 @@ avalanchego_node_json:
   snow-sample-size: 3
   snow-quorum-size: 2
   snow-mixed-query-num-push-vdr: 2
+  http-allowed-hosts: "*"
```
Comment (leopaul36) [inline inventories/local/group_vars/avalanche_nodes.yml:23]: Do we really need to override this since query with an empty `Host` header are accepted?

**Review by leopaul36 — APPROVED** [review]

File: inventories/local/group_vars/avalanche_nodes.yml (hunk 1 above)
Comment (Nuttymoon) [reply inventories/local/group_vars/avalanche_nodes.yml:23]: If we don't override this, the `Node security` tutorial won't work as the node won't accept incoming connections to `validator01.ash.local`

**Pull request merged**